metrics/registry.go

```diff
@@ -1,19 +1,50 @@
 package metrics
 
+var Reg *Registry
+
+const (
+	defaultMetricPort int = 9999
+	COUNTER           int = 0
+	GAUGE             int = 1
+	SUMMARY           int = 2
+	UNTYPED           int = 3
+	HISTOGRAM         int = 4
+)
+
+//generic metric structure
+type Metric struct {
+	MetricType  int
+	MetricName  string
+	MetricValue int64
+}
```

Comment:
struct itself has the name metric, we can drop "metric" from the values

```suggestion
type Metric struct {
	Type  MetricType
	Name  string
	Value int64
}
```